### PR TITLE
CSS Restructuring

### PR DESCRIPTION
**Pull request by jonahchoi**
#150 Restructured CSS architecture using global modules and TS declaration files to enable type-safe usage. 

1. `globals.css` is only for common variables and styling that applies to ALL elements 
2. Any reusable component styling (like buttons, inputs, etc) will be in a global module.
3. Component modules will be located within the component folder.
4. Styles will be imported as `$module_name`. Ex: `import $button from "@/styles/Button.module.css"`
5. Use the `type-css` npm command to automatically create declaration files for css, and the `type-css:check` command to verify these files are correct.

Co-authored-by: Hannah McGowen @hannahmcg 
Co-authored-by: Patrick Choi @patrickwchoi

### CODE REVIEW

**Review by codetheweb — COMMENTED** [review]
the `.module.css.d.ts` files add a lot of noise, what do you think about instead adding them to `.gitignore` (not checking them in) and adding a npm postinstall hook to generate them?

**Comment by jonahchoi** [comment]
@codetheweb I really wanted to shove them all into their own separate folder, but couldn't find a way for TS to recognize them without being in the same folder as the base css file. 

One downside I could think of is not being able to run type-css:check automatically on PRs, but I think its possible for the github action runner to create the files and then verify types are used correctly. I'll look into it.

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -12,10 +12,10 @@ jobs:
         with:
           node-version: 18
           cache: "npm"
-      - name: Install dependencies
+      - name: Install dependencies & generate css types
         run: npm i --ci
       - name: Check types
-        run: npm run type-check
+        run: npm run type:check
```
Comment (codetheweb) [inline .github/workflows/pr.yml:18]: I think this is moot if they're not being checked in and generated within the same workflow?

**Review by codetheweb — COMMENTED** [review]
nice

File: .github/workflows/pr.yml (hunk 1 above)
Comment (jonahchoi) [reply .github/workflows/pr.yml:18]: The css types are generated within the workflow, then the `type:check` (which is the regular typescript check) command will check for type errors throughout the repo. This way it'll effectively see if an imported css module is using an invalid property. 

The `type-css:check` command will still be available for us to use locally, in order to double check that our declaration files are updated.

Comment (codetheweb) [reply .github/workflows/pr.yml:18]: oh my bad, read too quickly

**Review by codetheweb — APPROVED** [review]

**Pull request merged**